### ISSUE TITLE
Accessability issues for website

### ISSUE THREAD

**Issue by adacable** [computeranonymous/computer#49]
Started work on already with the black and white version, but we need to add [ARIA roles](http://www.w3.org/TR/wai-aria/roles) and a large text version as a minimum as far as I can see. 

Any other options people would like adding, while we're at fixing these?

(edit: site to website to avoid confusion)

**Comment by drtortoise** [comment]
does there need to be a large text version of the site when "ctr +" / "cmd +" works for increasing the text size?

**Comment by public** [comment]
Given that the website basically just emulates a text file we could probably just set the font size to (e.g.) 150%. Large text is in these days right?

I'm not sure I get how ARIA will help the page given home simple it is but I've not actually played with ARIA or anything that uses it much. What will it do?

**Comment by tef** [comment]
I'm thinking bumping up the default font size would be a sensible thing.

**Comment by tef** [comment]
As well as making the white-on-black/black-on-white option in inverted colours

**Comment by AgentAntelope** [comment]
@tef regarding white on black/ black on white, I can't figure out how to change back from white on black once I've changed. Bug?

**Comment by adacable** [comment]
@AgentAntelope Indeed, it is. It was a ten second hack to get something out there. I can think of a way to do it in pure css, but I may decide to implement it in JS, especially if we add a text size button as well.

On the text size issue, yes, we can, but there is little/no loss in implementing it and shows we're trying our hardest to do outreach.

**Comment by tef** [comment]
I'd rather have "use Ctrl +/Ctrl -" to change text size. which I think is recommended by RNIB

**Comment by decklin** [comment]
Since there's some discussion of text size here, I'll comment instead of making a new issue:

In the meta viewport tag there is currently `user-scalable=false`. I think this could be bad for people on mobile who need the text to be larger and don't have some other kind of solution set up. I tried to pinch-zoom the site this morning on the bus and it didn't work.

I am not good at frontend computer so I don't know what the right thing is. Maybe `initial-scale`?

**Comment by adacable** [comment]
@ntlk did that tag. It's good from a standard scrolling POV(stops people sliding around from side to side) but not sure how it is accessibility wise. I've no idea how mobile and accessibility intersect, as we can't do ctrl+ and ctrl- on mobile(to my knowledge)

**Comment by drtortoise** [comment]
three finger zoom still works
http://www.456bereastreet.com/archive/201105/ios_tip_how_to_zoom_on_web_pages_that_have_disabled_user_zoom/
http://www.456bereastreet.com/archive/201106/the_ios_zoom_setting_disables_maximum-scale1_and_user-scalableno/

**Comment by russss** [comment]
What deep magic is three finger zoom!? It doesn't seem to work on Android
at any rate. Which fingers do I use.

**Comment by adacable** [comment]
On 30/09/13 20:53, Russ Garrett wrote:

> What deep magic is three finger zoom!? It doesn't seem to work on Android
> at any rate. Which fingers do I use.
> On 30 Sep 2013 20:30, "L. Sandvik" notifications@github.com wrote:
> 
> > three finger zoom still works
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/computeranonymous/computer/issues/49#issuecomment-25395856
> > .
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/computeranonymous/computer/issues/49#issuecomment-25398828
> 
> So people can scale it, It's just not standard? That seems to be the
> best solution all round, to me at least.

It doesn't impact able peoples usability and people with disabilities
which mean they need bigger text will know about three finger zoom and
it's opposite on driod(assuming it exists)

## 

---

Thomas Walpole, Maker
drcable.org @drcabl3 drcabl3.tumblr.com

**Comment by russss** [comment]
Oh wow, I've just discovered Android accessibility gestures and they're awesome.

**Comment by decklin** [comment]
Oh, hey, I had no idea about the three finger zoom thing. Thank you!

To be honest I probably won't leave it turned on but I feel better knowing that people have options.

**Comment by tef** [comment]
I'm leaving this open, but it seems vague and open-ended. Can we have some detail about what needs to be done before we can make this closed.

**Comment by adacable** [comment]
I considered ARIA roles expensive before but now we're in a generator I'll implement them. 

I propose we also add ARIA markup to the style guide, and once we're compliant on all pages we close it? 

We can re-open if we get complaints. I'll act as somone to @/email/irc if people are unsure about ARIA roles if you feel we need somone.

**Comment by tef** [comment]
Judging from what i've read about ARIA roles, and our website, which is simply links and text, I am unsure if there is much semantic information to add (HTML5 microdata might be useful for events), as we won't do much in the way of silly js or html stuff, and much to be gained. I'm happy to merge a PR in but I don't consider it a blocker for accessibility, unlike the Black on White not persisting

**Comment by tef** [comment]
Closing. We can open new PRs for ARIA roles, and nothing is blocking accessibility yet, that isn't already in another issue.
